lib/rrd.ml

```diff
@@ -545,9 +545,17 @@ let from_xml input =
         let rows = Array.length data in
         let cols = try Array.length data.(0) with _ -> -1 in
         let db = Array.init cols (fun _ -> Fring.make rows nan) in
+        (* CA-325844 Do not allow out-of-range values in RRAs *)
+        let sanitize ds value =
+          let value = float_of_string value in
+          if (Utils.isnan value) || value < ds.ds_min || value > ds.ds_max then
```

Comment:
`(...)` are not required. Could `float_of_string` fail?

Reply:
yes, it can fail:
```
# float_of_string "hello";;
Exception: Failure "float_of_string".
```

I'll double check all the conversions around here as the ones I've seen don't account for failures.

Reply:
On second thought, if the serialization of one of the other fields fails I'm inclined to think that it's preferable that the serialization of the whole rrd fails.

This is because 1) they are immutable fields and 2) they affect how new values are calculated.

Better treat the whole rrd as corrupted and start anew than risk having strange behaviour.